requests/adapters.py

```diff
@@ -392,7 +392,16 @@ def send(self, request, stream=False, timeout=None, verify=True, cert=None, prox
                         low_conn.send(b'\r\n')
                     low_conn.send(b'0\r\n\r\n')
 
-                    r = low_conn.getresponse()
+                    # Control timeout manually because we do not use
+                    # 'urlopen' but internal connection object.
+                    if hasattr(low_conn, 'sock'):
+                        low_conn.sock.settimeout(timeout.read_timeout)
+
+                    try:
+                        r = low_conn.getresponse()
+                    except (socket.error, socket.timeout) as err:
+                        conn._raise_timeout(err, url, timeout.read_timeout)
```

Comment:
I'm not quite a fan of using a private method here. I'm otherwise, okay with the rest of this change.